src/main/java/Duke.java

```diff
@@ -1,10 +1,289 @@
+import java.util.Scanner;
+import java.lang.String;
+import java.util.ArrayList;
+import java.io.*;
+
 public class Duke {
+
+    public static final String line = "____________________________________________________________\n";
+    public static final int MAX_NUM_OF_TASKS = 100;
+    public static final int TODO_POS = 5;
+    public static final int BY_POS = 4;
+    public static final int DEADLINE_POS = 9;
+    public static final int AT_POS = 4;
+    public static final int EVENT_POS = 6;
+    public static final int TYPE_POS = 4;
+    public static final int TASK_POS = 10;
+
     public static void main(String[] args) {
+
+        greeting();
+        ArrayList<Task> tasks = getDatafromFile("tasks");
+
+        if (tasks.size() == 0) {
+            System.out.println("No tasks can be loaded.");
+        } else {
+            System.out.println("Tasks have been loaded.");
+        }
+
+
+        Scanner input = new Scanner(System.in);
+        while (true) {
+            String inputString = input.nextLine();
+            if (inputString.equals("bye")) {
+                saveDataToFile("tasks", tasks);
+                exit();
+                break;
+            } else if (inputString.equals("list")) {
+                try {
+                    printList(tasks);
+                } catch (DukeException d) {
+                    System.out.println(line
+                            + d.getMessage() + "\n" + line);
+                }
+            } else if (inputString.contains("done") == true) {
```

Comment:
There is no need to explicitly check if it is equal to true because the contains method returns a boolean value. You can change this to:
```
else if (inputString.contains("done")) {
}
```